android/java/org/chromium/chrome/browser/crypto_wallet/activities/AddAccountActivity.java

```diff
@@ -33,23 +38,42 @@
 import java.io.BufferedReader;
 import java.io.InputStream;
 import java.io.InputStreamReader;
+import java.util.Arrays;
 
 public class AddAccountActivity extends BraveWalletBaseActivity {
     public static final String ACCOUNT = "account";
     private static final String TAG = "AddAccountActivity";
 
+    private static final int FILECOIN_MAINNET_POSITION = 0;
+    private static final int FILECOIN_TESTNET_POSITION = 1;
+
     private String mAddress;
     private String mName;
+    private String mKeyringId;
     private boolean mIsUpdate;
     private boolean mIsImported;
     private EditText mPrivateKeyControl;
     private EditText mAddAccountText;
+    private EditText mImportAccountPasswordText;
+    private Spinner mFilecoinNetworkSpinner;
     private static final int FILE_PICKER_REQUEST_CODE = 1;
     private CryptoAccountTypeInfo mCryptoAccountTypeInfo;
     private WalletModel mWalletModel;
+    @FilecoinNetworkType
+    private String mSelectedFilecoinNetwork;
 
-    public AddAccountActivity() {
-        mIsUpdate = false;
+    @Override
+    protected void onPreCreate() {
```

Comment:
Why onPreCreate is used intead of onCreate here?

Reply:
The `onPreCreate` method is a callback exposed by Chromium, and is the right callback to use when checking an `Intent`. 
This is because there are tons of optimizations executed by the Chromoium stack, and the order of the callbacks (`finishNativeInitialization` for this specific case, as it uses the data extracted from the intent) may be different depending on many factors (device type, Android version, memory available and what not).

## TL;TD
The callback `finishNativeInitialization` may be fired before or after `triggeredLayoutInflation` (and `triggeredLayoutInflation` is one of the callback executed from `onCreate`). 

Loading the native libraries usually happens before triggering the layout inflation but only for regular startup. For instant start instead the layout inflation is triggered before the native library initialization.

### Regular startup vs Instant startup
The logic is handled by Chromium and is based on many factors, but to delve more into the topic this method can be used as a starting point: https://source.chromium.org/chromium/chromium/src/+/main:chrome/android/java/src/org/chromium/chrome/browser/init/AsyncInitializationActivity.java;drc=adb8fbb4e4d50842f10429c613231915b5225a0e;l=172